subprojects/core/src/integTest/groovy/org/gradle/api/tasks/IncrementalTasksIntegrationTest.groovy

```diff
@@ -60,18 +61,16 @@ class IncrementalTasksIntegrationTest extends AbstractIntegrationSpec {
 
             incrementalExecution = inputs.incremental
 
-            inputs.outOfDate { change ->
+            inputs.getChanges(inputDir).each { change ->
                 if (change.added) {
                     addedFiles << change.file
-                } else {
+                } else if (change.modified) {
                     changedFiles << change.file
+                } else {
+                    removedFiles << change.file
```

Comment:
This would look much better with a switch, no?

Reply:
I am using a switch now, though I don't see much improvement.